### PR TITLE
Assign grid belgium

### PR DESCRIPTION
**Pull request by damianooldoni**
This PR adds pipeline `assign_grid.Rmd` to `./src/belgium/`. In this pipeline:
1. Connection to database created in pipeline `create_db.Rmd` is established
2. latitude, longitude and coordinate uncertainty are loaded in memory
3. Default uncertainty 1000 meter is assigned to occurrences without uncertainty (value written to db as well)
4. The occurrence coordinates are transformed to Lambert (CRS of Belgian grid)
5. The occurrences are assigned randomly in the circle with radius equal to uncertainty
6. The grid cell where the occurrence falls in found and saved in column `eea_cell_code`
7. Column in 6. is written in database
8. Preview 
9. Connection to database closed.

IMPORTANT: if you still didn't start to review PR #5 , please ignore it as you can review this one and get two pipelines at the price of one :+1:

### CODE REVIEW

File: src/belgium/create_db.Rmd [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+## Get CSV file from GBIF
+
+We download the occurrences from GBIF, based on the key got in `download.Rmd`:
+
+```{r get_occ_file}
+key <- "0044021-181108115102211"
+zip_filename <- paste0(key, ".zip")
+if (!file.exists(here("data", "raw", zip_filename))) {
+  occ <- occ_download_get(
```
Comment (peterdesmet) [inline src/belgium/create_db.Rmd]: The variable `occ` is required in the next step, but is only created when the file does not yet exist. I will fix this.

File: src/belgium/create_db.Rmd [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+Open connection to database:
+
+```{r open_connection_to_db}
+sqlite_occ <- dbConnect(SQLite(), dbname = sqlite_path)
+```
+
+And fill it with occurrences from text file. This step reads the large occurrence file in chunks and transfer them in the sqlite file. This step can take long the first timeyou run it:
+
+```{r fill_sqlite_file}
```
Comment (peterdesmet) [inline src/belgium/create_db.Rmd:155]: I got the error:

>  Error in connection_import_file(conn@ptr, name, value, sep, eol, skip) : 
  RS_sqlite_import: /Users/peter_desmet/Coding/Repositories/trias-project/occ-processing/data/raw/0044021-181108115102211_occurrence.txt line 3637202 expected 237 columns of data but found 226

**Review by peterdesmet — CHANGES_REQUESTED** [review]
Had to stop review of create_db because of error.

File: src/belgium/create_db.Rmd (hunk 1 above)
Comment (damianooldoni) [reply src/belgium/create_db.Rmd]: Fixed here: 481a5d4ac3c4e5ad365467b7fb9a6a943f07b9ae

File: src/belgium/create_db.Rmd (hunk 2 above)
Comment (damianooldoni) [reply src/belgium/create_db.Rmd:155]: I run the pipeline again and I got no errors back. 
Thanks to the error report I could extract the lines around 3637202 with Windows-based text editor called [LogExpert](http://www.log-expert.de/)). I extracted the lines from 3637200 to 3637205 included (= 6 lines) and saved in [data/interim/bug_PR_6_discussion_r266808445.txt](https://github.com/trias-project/occ-processing/blob/assign-grid-belgium/data/interim/bug_PR_6_discussion_r266808445.txt). 
I don't see any particular difference in line 637202. They are all machine observations of gulls (_Larus_).
 I would first try running the pipeline again :crossed_fingers: 
If still not working, try to read the file I mentioned above in R. 
If nothing anomalous found, we can further check in your local MAC OS X settings.

Comment (damianooldoni) [reply src/belgium/create_db.Rmd:155]: Can you also run the commando `devtools::session_info()` in R and let me know which version of RSQLite you have installed on your Mac? Thanks.

Comment (peterdesmet) [reply src/belgium/create_db.Rmd:155]: Same bug.

<details>
<summary>Session info:</summary>

```r
> devtools::session_info()
─ Session info ────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       macOS Mojave 10.14.3        
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/Brussels             
 date     2019-04-12                  

─ Packages ────────────────────────────────────────────────────────────
 package     * version    date       lib source                    
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.5.2)            
 backports     1.1.3      2018-12-14 [1] CRAN (R 3.5.0)            
 bit           1.1-14     2018-05-29 [1] CRAN (R 3.5.0)            
 bit64         0.9-7      2017-05-08 [1] CRAN (R 3.5.0)            
 blob          1.1.1      2018-03-25 [1] CRAN (R 3.5.0)            
 broom         0.5.1      2018-12-05 [1] CRAN (R 3.5.0)            
 callr         3.1.1      2018-12-21 [1] CRAN (R 3.5.0)            
 cellranger    1.1.0      2016-07-27 [1] CRAN (R 3.5.0)            
 cli           1.1.0      2019-03-19 [1] CRAN (R 3.5.2)            
 colorspace    1.4-0      2019-01-13 [1] CRAN (R 3.5.2)            
 crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)            
 data.table    1.12.0     2019-01-13 [1] CRAN (R 3.5.2)            
 DBI           1.0.0.9001 2019-04-12 [1] Github (r-dbi/DBI@069df06)
 desc          1.2.0      2018-05-01 [1] CRAN (R 3.5.0)            
 devtools      2.0.1      2018-10-26 [1] CRAN (R 3.5.1)            
 digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.0)            
 dplyr       * 0.8.0.1    2019-02-15 [1] CRAN (R 3.5.2)            
 forcats     * 0.3.0      2018-02-19 [1] CRAN (R 3.5.0)            
 fs            1.2.6      2018-08-23 [1] CRAN (R 3.5.0)            
 generics      0.0.2      2018-11-29 [1] CRAN (R 3.5.0)            
 geoaxe        0.1.0      2016-02-19 [1] CRAN (R 3.5.0)            
 ggplot2     * 3.1.0      2018-10-25 [1] CRAN (R 3.5.0)            
 glue        * 1.3.1      2019-03-12 [1] CRAN (R 3.5.2)            
 gtable        0.2.0      2016-02-26 [1] CRAN (R 3.5.0)            
 haven         2.0.0      2018-11-22 [1] CRAN (R 3.5.0)            
 here        * 0.1        2017-05-28 [1] CRAN (R 3.5.0)            
 hms           0.4.2      2018-03-10 [1] CRAN (R 3.5.0)            
 httr          1.4.0      2018-12-11 [1] CRAN (R 3.5.1)            
 jsonlite      1.6        2018-12-07 [1] CRAN (R 3.5.0)            
 knitr         1.21       2018-12-10 [1] CRAN (R 3.5.1)            
 lattice       0.20-38    2018-11-04 [1] CRAN (R 3.5.2)            
 lazyeval      0.2.1      2017-10-29 [1] CRAN (R 3.5.0)            
 lubridate     1.7.4      2018-04-11 [1] CRAN (R 3.5.0)            
 magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)            
 memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)            
 modelr        0.1.2      2018-05-11 [1] CRAN (R 3.5.0)            
 munsell       0.5.0      2018-06-12 [1] CRAN (R 3.5.0)            
 nlme          3.1-137    2018-04-07 [1] CRAN (R 3.5.2)            
 oai           0.2.2      2016-11-24 [1] CRAN (R 3.5.0)            
 pillar        1.3.1      2018-12-15 [1] CRAN (R 3.5.0)            
 pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.5.0)            
 pkgconfig     2.0.2      2018-08-16 [1] CRAN (R 3.5.0)            
 pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.0)            
 plyr          1.8.4      2016-06-08 [1] CRAN (R 3.5.0)            
 prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)            
 processx      3.2.1      2018-12-05 [1] CRAN (R 3.5.0)            
 ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.0)            
 purrr       * 0.3.2      2019-03-15 [1] CRAN (R 3.5.2)            
 R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)            
 Rcpp          1.0.1      2019-03-17 [1] CRAN (R 3.5.2)            
 readr       * 1.3.1      2018-12-21 [1] CRAN (R 3.5.0)            
 readxl        1.1.0      2018-04-20 [1] CRAN (R 3.5.0)            
 remotes       2.0.2      2018-10-30 [1] CRAN (R 3.5.0)            
 rgbif       * 1.1.0.9311 2019-02-26 [1] local                     
 rgeos         0.4-2      2018-11-08 [1] CRAN (R 3.5.0)            
 rlang         0.3.4      2019-04-07 [1] CRAN (R 3.5.2)            
 rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)            
 RSQLite     * 2.1.1      2018-05-06 [1] CRAN (R 3.5.0)            
 rstudioapi    0.8        2018-10-02 [1] CRAN (R 3.5.0)            
 rvest         0.3.2      2016-06-17 [1] CRAN (R 3.5.0)            
 scales        1.0.0      2018-08-09 [1] CRAN (R 3.5.0)            
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.0)            
 sp            1.3-1      2018-06-05 [1] CRAN (R 3.5.0)            
 stringi       1.4.3      2019-03-12 [1] CRAN (R 3.5.2)            
 stringr     * 1.4.0      2019-02-10 [1] CRAN (R 3.5.2)            
 testthat      2.0.1      2018-10-13 [1] CRAN (R 3.5.0)            
 tibble      * 2.1.1      2019-03-16 [1] CRAN (R 3.5.2)            
 tidyr       * 0.8.2      2018-10-28 [1] CRAN (R 3.5.0)            
 tidyselect    0.2.5      2018-10-11 [1] CRAN (R 3.5.0)            
 tidyverse   * 1.2.1      2017-11-14 [1] CRAN (R 3.5.0)            
 usethis       1.4.0      2018-08-14 [1] CRAN (R 3.5.0)            
 whisker       0.3-2      2013-04-28 [1] CRAN (R 3.5.0)            
 withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.0)            
 xfun          0.4        2018-10-23 [1] CRAN (R 3.5.0)            
 xml2          1.2.0      2018-01-24 [1] CRAN (R 3.5.0)            
 yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.0)            

[1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
> 
```
</details>

Comment (peterdesmet) [reply src/belgium/create_db.Rmd:155]: Ok, my file is truncated when unzipping, which caused the file to abruptly end at 3637202:

```r
In unzip(zipfile = download_file, files = "occurrence.txt", exdir = here("data",  :
  possible truncation of >= 4GB file
```

Comment (peterdesmet) [reply src/belgium/create_db.Rmd:155]: It did work by using `unzip filename` in bash.

**Comment by peterdesmet** [comment]
Was able to run `create_db.Rmd` successfully.

**Comment by peterdesmet** [comment]
And `assign_grid.Rmd` as well.

**Review by peterdesmet — APPROVED** [review]

**Pull request merged**